### DISCUSSION TITLE
how to set "cmd-w"(command + w ) to select word like IDEA keymap

### DISCUSSION THREAD

**Discussion by aohanhongzhi** [zed-industries/zed#10172] [Feature Requests]
![image](https://github.com/zed-industries/zed/assets/37319319/698df0e1-d976-4363-8686-6284d703619d)

**Comment by JosephTLyons** [comment]
Hmm, that's strange. I'm not finding anything in our default keymap that suggests we have a way to select the word through the keyboard (same as double click on a word in the editor). I'll leave it to someone else on the team to confirm my findings. We might have to add code to make this possible.

**Reply by JosephTLyons** [reply]
Ok, Max reminded me that `SelectLargerSyntaxNode` exists - that should work.

```json
{
  "context": "Editor",
  "bindings": {
    "alt-shift-w": "editor::SelectLargerSyntaxNode"
  }
},
```

**Reply by stiang** [reply]
Annoyingly, the cursor gets positioned at the start of the selected text, which is the opposite of what I want and how most other editors do it (I think? I have used this in at least VSCode/Cursor/Windsurf, Sublime Text and probably other without ever being surprised by where the cursor gets positioned). 

Is there any way to get that behaviour? I’ve tried using `workspace::SendKeystrokes` to combine the SelectLargerSyntaxNode command with moving to the end of the word, but the selection gets cancelled by the move.
